converter/generate_image.py

```diff
@@ -1,13 +1,57 @@
 from flask import current_app, Blueprint, send_file, flash, redirect
 from music21 import *
+from zipfile import ZipFile
+import os
 bp = Blueprint("get_image", __name__)
 
 #download result image
 @bp.route('/get-image/')
 def generate_image():
     try:
-        return send_file(converter.parse('musicxmlCache/ActorPreludeSample.musicxml').write('musicxml.png'), 
-            attachment_filename="result.png" ,as_attachment=True, cache_timeout=0)
+        zipObj = ZipFile(current_app.config['OUTPUT_ZIP'], 'w')
+        a = converter.parse('musicxmlCache/ActorPreludeSample.musicxml').write('musicxml.png')
+        
+        if a[-6:] == '-1.png':
```

Comment:
This whole section is unfortunately poorly written because there is numerous hard-coded numbers without comments and explanations. 

I can read through it and understand that this is to handle the case when there are different total number of images, but why things have to be done this way? 

There are still questions that I'd like to ask:

1) What is a[:-6], a[:-7} correspond to? The way that it is written seems fragile to me;
2) So is there a total size cap for this logic? Looks to me there are (up to 1000)? Shouldn't there be some error handling if an unexpected case shows up? 
3) Where are the tests?????

Please address these issues and add sufficient comments regarding to what are the cases this is supporting, and what not. 

Once again, please add sufficient tests before piling on new features.

Reply:
will add more tests;

because now music21 generate files in `/tmp/music21` the filename follows the rules:
- 8 random characters(exp. `/tmp/music21/8jo2o_qz.png`)
- if there are multiple files, it goes:
            -if number of images < 10 : 
```
/tmp/music21/8jo2o_qz-1.png
/tmp/music21/8jo2o_qz-2.png
...
```
            -if 10 <= number of images < 100 : 
```
/tmp/music21/8jo2o_qz-01.png
/tmp/music21/8jo2o_qz-02.png
...
```
            -if 100 <= number of images <= 999 : 
```
/tmp/music21/8jo2o_qz-001.png
/tmp/music21/8jo2o_qz-002.png
...
```
when image numbers exceeds 999, `music21` itself will return an error `'png file of xml not found. Or file >999 pages?')`

Reply:
and `write()` method returns the filepath, or the firsr image filepath if there are multiple ones. So I check the path returned by looking at the last several characters.

Reply:
Yes, please clarify the cases here in the comment, and add tests correspondingly. Furthermore, make sure in music21 there is document clarifying the file length prefix (8 as you stated). What I am hoping to avoid is that there are cases that this prefix length vary. 

What you can also do is to search for "-1.png", "-01.png" and "-001.png" as suffix from the path, so you can avoid a[:-6], a[:-7} at stuff. I think this will be better as it doesn't depend on the prefix. 

Last but not least, you need to catch the error thrown by music21 when there is more than 1000 and report that to the users.

Reply:
length should not matter, because I am slicing the file path backward so it is guaranteed to get the suffix. 
Error message for more than 1000 pages will be thrown to users.